### PR TITLE
csplit: first implementation

### PR DESCRIPTION
**Pull request by mkindahl**
A first implementation of `csplit`.

### CODE REVIEW

**Comment by mkindahl** [comment]
Sorry about the time, not able to spend as much time as I intended. There are still things that I think can be improved in the code. Please note that the added tests need to be executed manually: there can be differences that does not indicate an issue.

**Comment by mkindahl** [comment]
Something seems to be wrong with the job to build using the beta compiler. Looking at the output it looks like it is stalling when retrieving the cache information:

```
...
creating directory /home/travis/.cargo[0m
adding /home/travis/build/GrayJack/coreutils/target to cache[0m
creating directory /home/travis/build/GrayJack/coreutils/target[0m
adding /home/travis/.rustup to cache[0m
creating directory /home/travis/.rustup[0m
adding /home/travis/.cache/sccache to cache[0m
creating directory /home/travis/.cache/sccache[0m


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

I have tried to build it locally with both nightly, stable, and beta, and it works fine.

File: csplit/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "csplit"
+version = "0.1.0"
+authors = ["Mats Kindahl <mats.kindahl@gmail.com>"]
+build = "build.rs"
+edition = "2018"
+
+[dependencies]
+clap = { version = "^2.33.0", features = ["yaml", "wrap_help"] }
+regex = "1.3.1"
```
Comment (GrayJack) [inline csplit/Cargo.toml]: Please, allow other versions of regex using the `^` at the start of the version number

File: csplit/src/csplit.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        help: Do not remove output files on error.
+    - suppress:
+        long: suppress-matched
+        short: x
+        help: Suppress lines that match a PATTERN.
+    - digits:
+        long: digits
+        short: n
+        value_name: DIGITS
+        help: Use the given number of digits for output file name. The default is 2.
```
Comment (GrayJack) [inline csplit/src/csplit.yml]: You can add a default field here with the value. It will also be shown in the help

File: csplit/src/csplit.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    - prefix:
+        long: prefix
+        short: f
+        value_name: PREFIX
+        help: Prefix to use for written files. The default is 'xx'.
+    - suffix-format:
+        long: suffix-format
+        short: b
+        value_name: FORMAT
+        help: Format to use for the file suffix. The default is '%02d'.
```
Comment (GrayJack) [inline csplit/src/csplit.yml]: You can add a default field here with the value. It will also be shown in the help

File: csplit/src/csplit.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+after_help: |-
+  PATTERN can be any of:
+    INTEGER            copy lines up to line number
+    /REGEXP/[OFFSET]   copy lines up to line matching REGEXP
+    %REGEXP%[OFFSET]   skip lines up to line matching REGEXP
+    {INTEGER}          repeat preceeding pattern INTEGER times
+    {*}                repeat preceeding pattern indefinitely
+
+  If an OFFSET is given it should be an integer, either positive or
+  negative. An offset without sign is assumed to be positive.
```
Comment (GrayJack) [inline csplit/src/csplit.yml]: I'm not sure if this fits best here or as `long_help` for `PATTERN`. It was ok in both for me.

Try both and give a feedback, please.

**Comment by GrayJack** [comment]
> There are still things that I think can be improved in the code, but I can take them as separate PRs

Can you open issues about each one so we can easily keep track of them?

**Comment by mkindahl** [comment]
> > There are still things that I think can be improved in the code, but I can take them as separate PRs
> 
> Can you open issues about each one so we can easily keep track of them?

Of course. :)

File: csplit/src/csplit.yml (hunk 4 above)
Comment (mkindahl) [reply csplit/src/csplit.yml]: `long_help` looked better.

File: csplit/src/csplit.yml (hunk 3 above)
Comment (mkindahl) [reply csplit/src/csplit.yml]: Done.

File: csplit/src/csplit.yml (hunk 2 above)
Comment (mkindahl) [reply csplit/src/csplit.yml]: Done.

File: csplit/Cargo.toml (hunk 1 above)
Comment (mkindahl) [reply csplit/Cargo.toml]: Done.

**Comment by GrayJack** [comment]
Looks like I pushed a something that breaks 1.37 compatibility

**Pull request merged**